common/config/loader.go

```diff
@@ -46,66 +58,147 @@ const (
 // The loader first fetches the set of files matching
 // a pre-determined naming convention, then sorts
 // them by hierarchy order and after that, simply
-// loads the files one after another with the
-// key/values in the later files overriding the key/values
-// in the earlier files
+type loadOptions struct {
+	env             string
+	configDir       string
+	zone            string
+	configFilePath  string
+	useEmbeddedOnly bool
+	envMap          map[string]string
+}
+
+type loadOption func(*loadOptions)
+
+func WithEnv(env string) loadOption {
```

Comment:
nit: Wondering why you chose the functional options pattern vs exposing the `loadOptions` struct with all of the fields directly. Not blocking the PR as we have both patterns in the codebase, but it seems like it the alternative would be simpler.